### DISCUSSION TITLE
[riverpod_generator] how to override provider which generated from riverpod_generator

### DISCUSSION THREAD

**Discussion by drown0315** [rrousselGit/riverpod#2501] [Q&A] [answered]
how to override provider generated from riverpod_generator

**Comment by rrousselGit** [comment, marked as answer]
Like non-generated providers:

```dart
// For classes:
@riverpod
class Example extends _$Example {
  int build() => 0;
}

exampleProvider.overrideWith(() => Example());

// For functions:
@riverpod
int example(ExampleRef ref) => 0;

exampleProvider.overrideWith((ref) => 0);
```

**Reply by drown0315** [reply]
but if my provider type is family, how to do?

**Reply by artahc** [reply]
I have this question as well
How to override a family provider, but ignoring the params. Previously with non-generated family provider, we can override them with another provider while ignoring its params

**Reply by tonnyavery** [reply]
`overrideWith` accepts `Function()`, so ref is not available. Were there some changes that are not yet documented?

**Reply by rrousselGit** [reply]
@AAverin A ref is available when using the function syntax. When using the class syntax, you're expected to return a class instance. That class will have access to the ref.

**Reply by tonnyavery** [reply]
Odd, this doesn't seem to work. I will check if I am on the right version
```
await $.pumpWidgetAndSettle(ProviderScope(overrides: [
        habitsRepoProvider.overrideWith((ref) => ref.read(habitsRepoProvider))
      ], child: MyApp()));
```

`The argument type 'HabitsRepo Function(dynamic)' can't be assigned to the parameter type 'HabitsRepo Function()'. `

**Comment by marlonmartina** [comment]
Is there an update on this? I'm also trying to override an auto generated family provider, see implementation: 
```
ProviderContainer(
  overrides: <Override>[
    registerAppProvider(
      serviceUrl: "serviceUrl",
      sessionToken: "sessionToken",
      registerAppRequest: const RegisterAppRequest(
        encryptedAppIdentity: "encryptedAppIdentity",
      ),
    ).overrideWith(
      (provider) => const RegisterAppResponse(
        registrationToken: "registrationToken",
      ),
    ),
  ],
);

```
But the original provider is still being called. Seems like riverpod doesn't find this override in the provider scope.

**Reply by rrousselGit** [reply]
Update on what? The solution was given.

Your issue is likely elsewhere. Maybe you're using that container incorrectly

**Reply by marlonmartina** [reply]
My problem is with how family providers work. When we use them, the inputs we give to call the provider have to be exactly the same as what we set up in the override. This makes it tricky to mock them, especially in certain situations. What I'm trying to do is to have a way to say 'any()' for the arguments, so that no matter what inputs I provide, the scoped version of the provider will always be used.

**Reply by dustin-graham** [reply]
@rrousselGit  I see that version 3.0.0 generates the desired code. Is there a way to achieve this in v2.x or do we just need to wait for 3.0 to land? https://github.com/rrousselGit/riverpod/commit/06aeaf0b71ca1fe59facf9258e6a33d31c91ae4e
